### PR TITLE
Move Filter To ForEach

### PR DESCRIPTION
**Pull request by adamgfraser**

### CODE REVIEW

**Review by sideeffffect — DISMISSED** [review]
We seem to have lost `withFilter`. Otherwise the code looks fine. But what was the motivation, if you don't mind me asking?

**Comment by adamgfraser** [comment]
It allows the effect type to be different than the collection type.

**Review by sideeffffect — APPROVED** [review]
Cool

File: core/shared/src/main/scala/zio/prelude/ForEach.scala [hunk 1]
```diff
@@ -488,6 +570,8 @@ trait ForEachSyntax {
       F.sum(self)
     def toChunk(implicit F: ForEach[F]): Chunk[A]                                                               =
       F.toChunk(self)
+    def withFilter(f: A => Boolean)(implicit F: ForEach[F], I: IdentityEither[F], B: IdentityBoth[F]): F[A]     =
```
Comment (sideeffffect) [inline core/shared/src/main/scala/zio/prelude/ForEach.scala:573]: Maybe we can add the comment
```
// Provides support for filtering `F[A]` values in for comphrensions.
```

**Pull request merged**